### PR TITLE
Apply permission boundary only to users

### PR DESCRIPTION
**Pull request by kylelaker-ccpo**

### CODE REVIEW

**Comment by kylelaker-ccpo** [comment]
@zachclark-ccpo an interesting test to work on in #641 might be to assert that there isn't a Permission Boundary defined for some types of resources

File: lib/aspects/user-only-permission-boundary.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  constructor(boundary: iam.ManagedPolicy) {
+    this.policy = boundary;
+  }
+
+  visit(node: IConstruct): void {
+    if (!(node instanceof iam.User)) {
+      return;
+    }
+    iam.PermissionsBoundary.of(node).apply(this.policy);
+  }
```
Comment (juliusfitzhugh-ccpo) [inline lib/aspects/user-only-permission-boundary.ts:16]: When is this `visit` method actually being called, during `cdk deploy`? 

Also, I believe you mentioned it already, but want to make sure I understand, but here the users IAM gets an additional policy to invoke the `api-gw`, along with any other permissions (if any) that are present for that particular user?

**Review by juliusfitzhugh-ccpo — APPROVED** [review]
Added few questions to make sure I understand. 

The fix works and I am able to run the `lambdas` and `state machine` fine. I did notice the `api-gw` log did not report anything.

File: lib/aspects/user-only-permission-boundary.ts (hunk 1 above)
Comment (kylelaker-ccpo) [reply lib/aspects/user-only-permission-boundary.ts:16]: The CDK itself leverages the [Visitor pattern](https://refactoring.guru/design-patterns/visitor) through [Aspects](https://docs.aws.amazon.com/cdk/v2/guide/aspects.html). So we don't call `visit` but it gets applied by the Aspects framework.

Comment (jeffsegal-ccpo) [reply lib/aspects/user-only-permission-boundary.ts:16]: I think Julius was asking whether it the aspect was actually applied during `synth` or `deploy`.

Comment (juliusfitzhugh-ccpo) [reply lib/aspects/user-only-permission-boundary.ts:16]: Thanks for pointing out documentation that helps clear things up.

**Pull request merged**